src/script/components/MessagesList/Message/ContentMessage/asset/TextMessageRenderer.tsx

```diff
@@ -66,98 +66,49 @@ export const TextMessageRenderer: FC<TextMessageRendererProps> = ({
-        emailLink.setAttribute('tabindex', isCurrentConversationFocused ? '0' : '-1');
-      });
-    }
+      ? [...containerRef.current.querySelectorAll('.message-mention')]
+      : [];
 
     const handleKeyEvent = (event: KeyboardEvent, elementType: ElementType) => {
       if (isCurrentConversationFocused) {
         handleKeyDown(event, () => onMessageClick(asset, event, elementType));
       }
     };
 
-    const handleClickEmail = (event: Event) => {
-      return event.type === 'keydown'
-        ? handleKeyEvent(event as KeyboardEvent, 'email')
-        : onMessageClick(asset, event as MouseEvent, 'email');
-    };
-    emailLinks?.forEach(emailLink => {
-      events.forEach(eventName => {
-        emailLink.addEventListener(eventName, handleClickEmail);
-      });
-    });
-
-    const handleClickLink = (event: Event) => {
-      return event.type === 'keydown'
-        ? handleKeyEvent(event as KeyboardEvent, 'markdownLink')
-        : onMessageClick(asset, event as MouseEvent, 'markdownLink');
-    };
-
-    linkTargets?.forEach(msgLink => {
-      events.forEach(eventName => {
-        msgLink.addEventListener(eventName, handleClickLink);
-      });
-    });
-
-    const handleClickMention = (event: Event) => {
+    const handleMsgEvent = (event: Event) => {
       return event.type === 'keydown'
         ? handleKeyEvent(event as KeyboardEvent, 'mention')
         : onMessageClick(asset, event as MouseEvent, 'mention');
     };
 
-    msgMention?.forEach(mention => {
-      events.forEach(eventName => {
-        mention.addEventListener(eventName, handleClickMention);
-      });
-    });
-
-    return () => {
-      emailLinks?.forEach(emailLink => {
+    function addEventListener(elements: Element[], elementType: ElementType) {
+      elements?.forEach(element => {
         events.forEach(eventName => {
-          emailLink.removeEventListener(eventName, handleClickEmail);
+          element.addEventListener(eventName, handleMsgEvent);
         });
       });
+    }
 
-      linkTargets?.forEach(msgLink => {
+    function removeEventListener(elements: Element[], elementType: ElementType) {
+      elements?.forEach(element => {
         events.forEach(eventName => {
-          msgLink.removeEventListener(eventName, handleClickLink);
+          element.removeEventListener(eventName, handleMsgEvent);
         });
       });
+    }
 
-      msgMention?.forEach(mention => {
-        events.forEach(eventName => {
-          mention.removeEventListener(eventName, handleClickMention);
-        });
-      });
+    addEventListener(emailLinks, 'email');
+    addEventListener(markdownLinkTargets, 'markdownLink');
+    addEventListener(msgMention, 'mention');
+
+    return () => {
+      removeEventListener(emailLinks, 'email');
+      removeEventListener(markdownLinkTargets, 'markdownLink');
+      removeEventListener(msgMention, 'mention');
     };
   }, [onMessageClick, asset, isCurrentConversationFocused, containerRef]);
```

Comment:
```suggestion
  }, [onMessageClick, asset, isCurrentConversationFocused]);
```

Refs do not need to be added to react dependencies as they do not trigger re-renders